middleware/grpc_instrumentation.go

```diff
@@ -10,6 +10,7 @@ import (
 	"strconv"
 	"time"
 
+	"github.com/gogo/status"
```

Comment:
Can we not use gogo/status in the new code?

Reply:
This has been discussed via Slack. There is already a private method handling this problem in `httpgrpc.statusFromError()`. That method has been moved to the `grpcutil` package, exported and renamed to `grpcutil.ErrorToStatus()`.

Our proto files are converted to structs by using `gogo/protobuf`, so as far as this is the case, `dskit` should prefer `gogo/status` to `grpc/status` for consistency.